File: ts/1664/1572.ts
function diagonalSum(mat: number[][]): number {
  const len = mat.length
  let [sum, i, j] = [0, 0, len - 1]
  while(i < len) {
    sum += mat[i][i]
    if (i !== j) sum += mat[i][j]
    i += 1
    j -= 1
  }
  return sum
};

File: py3/1664/1572.py
from typing import List

class Solution:
    def diagonalSum(self, mat: List[List[int]]) -> int:
        n = len(mat)
        ans, i, j = 0, 0, n - 1
        while i < n:
            ans += mat[i][i]
            if i != j:
                ans += mat[i][j]
            i += 1
            j -= 1
        return ans